cmake/toolchain_llvm.cmake

```diff
@@ -11,13 +11,42 @@ set(CMAKE_SYSTEM_NAME Generic)
 
 set(LLVM_TAG llvm)
 
-set(CMAKE_C_COMPILER ${TOOLCHAIN_DIR}/clang)
-set(CMAKE_CXX_COMPILER ${TOOLCHAIN_DIR}/clang++)
-set(CMAKE_ASM_COMPILER ${TOOLCHAIN_DIR}/clang)
+set(CMAKE_C_COMPILER ${TOOLCHAIN_DIR}/bin/clang)
+set(CMAKE_CXX_COMPILER ${TOOLCHAIN_DIR}/bin/clang++)
+set(CMAKE_ASM_COMPILER ${TOOLCHAIN_DIR}/bin/clang)
 
-set(CMAKE_OBJCOPY ${TOOLCHAIN_DIR}/${LLVM_TAG}-objcopy)
-set(CMAKE_OBJDUMP ${TOOLCHAIN_DIR}/${LLVM_TAG}-objdump)
-set(CMAKE_AR ${TOOLCHAIN_DIR}/${LLVM_TAG}-ar)
+set(CMAKE_OBJCOPY ${TOOLCHAIN_DIR}/bin/${LLVM_TAG}-objcopy)
+set(CMAKE_OBJDUMP ${TOOLCHAIN_DIR}/bin/${LLVM_TAG}-objdump)
+set(CMAKE_AR ${TOOLCHAIN_DIR}/bin/${LLVM_TAG}-ar)
+
+# Disable ABI detection
+set(CMAKE_C_ABI_COMPILED "False")
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --target=riscv32-unknown-elf")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --target=riscv32-unknown-elf")
\ No newline at end of file
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --target=riscv32-unknown-elf")
```

Comment:
Why don't you
`add_compile_options(
--target=riscv32-unknown-elf
)`
instead? Seems more cmake-esque to me.

Reply:
Good point; this was there before, but changing makes sense.